### PR TITLE
Fix database strict mode issues

### PR DESCRIPTION
**Pull request by toanhminh0412**
This is optional. At some point, I was running my hub database in a strict mode and found some errors. This MR contains the fixes to those errors

### CODE REVIEW

**Comment by nkissebe** [comment]
This will need closer review as well

**Comment by nkissebe** [comment]
Change mariadb server mode config to 
`sql-mode = ""`

I'll consider the changes regardless, but you are likely to be chasing issues from this for a while

**Comment by toanhminh0412** [comment]
Oh yes, I figured that out later on and changed the mode. However, I thought this PR would be beneficial to those who want to run their database in a stricter mode. It's totally up to you what you want to do with this PR.

File: core/components/com_collections/models/item.php [hunk 1]
```diff
@@ -520,6 +520,11 @@ public function vote()
 	 */
 	public function store($check=true)
 	{
+        // id cannot be an empty string due to database constrain
```
Comment (nkissebe) [inline core/components/com_collections/models/item.php]: fix indentation to match

File: core/components/com_wiki/models/page.php [hunk 2]
```diff
@@ -638,6 +638,14 @@ public function save()
 			}
 
 			$this->set('path', implode('/', $path));
+		} else {
```
Comment (nkissebe) [inline core/components/com_wiki/models/page.php]: Put curly braces on line by themselves to maintain coding style such as it is

File: core/components/com_wiki/models/page.php [hunk 3]
```diff
@@ -638,6 +638,14 @@ public function save()
...
+			// Must set parent as 'parent' field requires an int in the database
+			$this->set('parent', 0);
+                       
+			// Must set a path as 'path' field doesn't have a default in the database
+			if (!$this->get('path')) {
```
Comment (nkissebe) [inline core/components/com_wiki/models/page.php]: Put curly braces on line by themselves

File: core/plugins/groups/calendar/calendar.php [hunk 4]
```diff
@@ -730,6 +731,11 @@ private function save()
 			$event['registerby'] = null;
 		}
 
+		// calender_id cannot be an empty string due to database constrain
+        if ($event['calendar_id'] === "") {
```
Comment (nkissebe) [inline core/plugins/groups/calendar/calendar.php]: Put curly braces on line by themselves and line up indentation to match surrounding code

**Comment by nkissebe** [comment]
We can go ahead and accept this, just fix the code styling as commented

File: core/components/com_collections/models/item.php (hunk 1 above)
Comment (toanhminh0412) [reply core/components/com_collections/models/item.php]: Fixed

File: core/plugins/projects/publications/publications.php [hunk 5]
```diff
@@ -1260,6 +1260,8 @@ public function createDraft()
 		$row = new \Components\Publications\Tables\Version($this->_database);
 		$row->publication_id = $this->_pid;
 		$row->title          = $row->getDefaultTitle($this->model->get('id'), $title);
+		$row->abstract           = "";
```
Comment (nkissebe) [inline core/plugins/projects/publications/publications.php]: line up = with other lines to maintain style

File: core/components/com_wiki/models/page.php (hunk 2 above)
Comment (toanhminh0412) [reply core/components/com_wiki/models/page.php]: Fixed

File: core/components/com_wiki/models/page.php (hunk 3 above)
Comment (toanhminh0412) [reply core/components/com_wiki/models/page.php]: Fixed

File: core/plugins/groups/calendar/calendar.php (hunk 4 above)
Comment (toanhminh0412) [reply core/plugins/groups/calendar/calendar.php]: Fixed

File: core/components/com_collections/models/item.php [hunk 6]
```diff
@@ -520,6 +520,11 @@ public function vote()
 	 */
 	public function store($check=true)
 	{
+        	// id cannot be an empty string due to database constrain
+		if ($this->get('id') === "") {
```
Comment (nkissebe) [inline core/components/com_collections/models/item.php]: Another curly brace that needs to be moved to seperate line

File: core/plugins/groups/calendar/calendar.php [hunk 7]
```diff
@@ -730,6 +731,12 @@ private function save()
 			$event['registerby'] = null;
 		}
 
+		// calender_id cannot be an empty string due to database constrain
+        if ($event['calendar_id'] === "")
```
Comment (nkissebe) [inline core/plugins/groups/calendar/calendar.php]: Indentation problems with this block

**Comment by toanhminh0412** [comment]
I wonder if the indentation stuffs will affect different text editors. I'm using VS Code and everything lines up correctly on my editor. Though it doesn't appear so in git

File: core/plugins/projects/publications/publications.php (hunk 5 above)
Comment (toanhminh0412) [reply core/plugins/projects/publications/publications.php]: Fixed

File: core/components/com_collections/models/item.php (hunk 6 above)
Comment (toanhminh0412) [reply core/components/com_collections/models/item.php]: Fixed

File: core/plugins/groups/calendar/calendar.php (hunk 7 above)
Comment (toanhminh0412) [reply core/plugins/groups/calendar/calendar.php]: Fixed

**Comment by nkissebe** [comment]
You'll need to match your VS Code editor settings to what the code is using which I believe is indent with tabs.

File: core/plugins/groups/calendar/calendar.php [hunk 8]
```diff
@@ -730,6 +731,12 @@ private function save()
 			$event['registerby'] = null;
 		}
 
+	// calender_id cannot be an empty string due to database constrain
```
Comment (nkissebe) [inline core/plugins/groups/calendar/calendar.php]: Indention appears one off. I think the issue is needing to use tabs.

Comment (toanhminh0412) [reply core/plugins/groups/calendar/calendar.php]: I'm using tab in VSCode

Comment (toanhminh0412) [reply core/plugins/groups/calendar/calendar.php]: ok, this should be fixed now?

Comment (toanhminh0412) [reply core/plugins/groups/calendar/calendar.php]: Hmm idk what went wrong 🤷

**Comment by nkissebe** [comment]
Sorry to nitpick on this one so much. Looks good now.

**Comment by toanhminh0412** [comment]
No worries, I have no problems with it. I was just concerned that the indentation will appear improperly for some text editors

**Pull request merged**